tests/python_functional/src/helpers/loggen/conftest.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) 2020 One Identity
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import pytest
+from pathlib2 import Path
+from psutil import TimeoutExpired
+
+import src.testcase_parameters.testcase_parameters as tc_parameters
+from src.common.blocking import wait_until_true
+from src.executors.process_executor import ProcessExecutor
+from src.syslog_ng.syslog_ng_paths import SyslogNgPaths
+
+
+class Loggen(object):
+
+    def __init__(self, testcase_parameters):
+        self.loggen_proc = None
+        self.target = None
+        self.port = None
+        self.parameters = None
+        self.loggen_stdout_path = Path(tc_parameters.WORKING_DIR, "loggen_stdout")
+        self.loggen_stderr_path = Path(tc_parameters.WORKING_DIR, "loggen_stderr")
+        self.instance_path = SyslogNgPaths(testcase_parameters).set_syslog_ng_paths("server")
+        self.loggen_bin_path = self.instance_path.get_loggen_bin()
+
+    def is_loggen_started(self):
+        if not self.loggen_stderr_path.exists():
+            return False
+
+        # loggen puts the count= messages to the stderr
+        f = open(self.loggen_stderr_path, "r")
+        result = "count=" in f.read()
+        f.close()
+        return result
+
+    def start(self, target, port, parameters):
+        if self.loggen_proc is not None:
+            return
+
+        self.target = target
+        self.port = port
+        self.parameters = parameters
+
+        self.loggen_proc = ProcessExecutor().start(
+            [self.loggen_bin_path] + self.parameters + [self.target, self.port],
+            self.loggen_stdout_path,
+            self.loggen_stderr_path,
+        )
+
+        wait_until_true(self.is_loggen_started)
+        return self.loggen_proc.is_running()
+
+    def stop(self):
+        if self.loggen_proc is None:
+            return
+
+        self.loggen_proc.terminate()
+        try:
+            self.loggen_proc.wait(4)
+        except TimeoutExpired:
+            self.loggen_proc.kill()
+
+        self.loggen_proc = None
+
+
+@pytest.fixture
+def loggen(testcase_parameters):
```

Comment:
As a fixture, the TC (and possibly different TCs running in parallel) can only start 1 `Loggen` at once. We might not want to use it as a fixture.

Reply:
in the current form we can start only one instance. it won't be a big work to make it possible to start multi instances. for this we have to prepare the stop function to stop a process with the given process id.

please note: the stop function is not called from test case scripts directly so the process ids can be stored internally in the loggen/conftest.py file. the stop shall iterate over all the internally stored process ids.

I'll do it later

Reply:
Or the users can import `Loggen` and create multiple objects from it. It is perfect in this form. Thanks!